features/ui/select/select.tsx

```diff
@@ -0,0 +1,141 @@
+import React, {
+  forwardRef,
+  useImperativeHandle,
+  useState,
+  ReactElement,
+} from "react";
+import styles from "./select.module.scss";
+
+type Option = {
+  value: string;
+  label: string;
+};
+
+type SelectBoxProps = {
+  options: Option[];
+  onChange: (value: string) => void;
+  placeholder: string;
+  disabled?: boolean;
+  icon?: ReactElement;
+  label?: string;
+  hint?: string;
+  errorText?: string;
+};
+
+export const SelectBox = forwardRef<
+  { setValue: (value: string) => void },
+  SelectBoxProps
+>(
+  (
+    { options, onChange, placeholder, disabled, icon, label, hint, errorText },
+    ref,
+  ) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const [selectedOption, setSelectedOption] = useState<Option | null>(null);
+
+    const handleOptionClick = (option: Option) => {
+      setSelectedOption(option);
+      onChange(option.value);
+    };
+
+    // This allows the parent component to control the value of the select box
+    useImperativeHandle(ref, () => ({
+      setValue: (value: string) => {
+        const option = options.find((option) => option.value === value);
+        setSelectedOption(option || null);
+      },
```

Comment:
With this the parent would be able to set the value on its child right? like

```
function Parent() {
  const select = useRef();
  return (
   <>
     <Select ref={select} />
     <button onClick={() => select.current.setValue(1)}>
      Set value
    </button>
  </>
  )
}
```

I'd consider this an anti pattern tbh. Communitcation from the parent to its children usually works via props and from a child to its parent via callbacks (like an onClick handler). So in this case it would be better add a `value` prop to the Select component so the parent can control the value. That's also referred to as a "controlled" component.